src/stores/useSocialMediaStore/useSocialMediaStore.spec.ts

```diff
@@ -0,0 +1,101 @@
+
+vi.mock('~services/api/social-media/social-media', () => ({
+  SocialMediaService: {
+    fetch: vi.fn().mockResolvedValue([
+      { icon: 'discord-icon', id: 'DISCORD_EXAMPLE_ID', name: 'Discord' },
+      { icon: 'twitter-icon', id: 'TWITTER_EXAMPLE_ID', name: 'Twitter' },
+    ]),
+  },
+}));
+
+vi.mock('~services/api/accounts/accounts', () => ({
+  AccountsService: {
+    fetchAll: vi.fn().mockResolvedValue([
+      {
+        avatar: 'https://example.com/image1.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 1,
+        socialMediaId: 'DISCORD_EXAMPLE_ID',
+        token: 'DISCORD_EXAMPLE_TOKEN_1',
+        userName: 'Discord User 1',
+        valid: true,
+      },
+      {
+        avatar: 'https://example.com/image2.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 14,
+        socialMediaId: 'TWITTER_EXAMPLE_ID',
+        token: 'TWITTER_EXAMPLE_TOKEN_14',
+        userName: 'Twitter User 14',
+        valid: true,
+      },
+    ]),
+  },
+}));
+
+/*
+The getAllAccounts call is necessary because although the getAllAccounts function is 
+defined in the store body, it is not automatically executed when the state object is 
+created and the behavior we want is to be called automatically when the store is initialized.
+*/
+
+describe('useSocialMediaStore', () => {
+  it('populates the socialMedias store from user accounts data', async () => {
+    const { result } = renderHook(() =>
+      useSocialMediaStoreHook.useSocialMediaStore()
+    );
+
+    await act(async () => {
+      await result.current.getAllAccounts();
+    });
+
+    const expectedMap = new Map([
+      [
+        'DISCORD_EXAMPLE_ID',
+        {
+          icon: 'discord-icon',
+          id: 'DISCORD_EXAMPLE_ID',
+          name: 'Discord',
+        },
+      ],
+      [
+        'TWITTER_EXAMPLE_ID',
+        {
+          icon: 'twitter-icon',
+          id: 'TWITTER_EXAMPLE_ID',
+          name: 'Twitter',
+        },
+      ],
+    ]);
+
+    expect(result.current.socialMedias).toEqual(expectedMap);
+  });
+
+  it('Checks if getAllAccounts correctly retrieves and validates social media accounts', async () => {
+    const updatedAccounts = [
+      { id: 1, socialMediaId: 'DISCORD_EXAMPLE_ID', valid: false },
+      { id: 14, socialMediaId: 'TWITTER_EXAMPLE_ID', valid: false },
+    ];
+
+    const { result } = renderHook(() =>
+      useSocialMediaStoreHook.useSocialMediaStore()
+    );
+
+    await act(async () => {
+      await result.current.getAllAccounts();
+    });
+
+    expect(result.current.accounts.data).toEqual(
+      expect.arrayContaining(
+        updatedAccounts.map((account) => expect.objectContaining(account))
+      )
+    );
+  });
```

Comment:
Nao entendi esse teste, ele testa oq?

Reply:
Seguinte, o estado `current.accounts.data ` 'e o "resultado" da  instância da classe MultiMap que armazena objetos do tipo StoreAccount, certo ? O formato de saida, apos a entrada de dados passar pela trecho abaixo:

`
    for (const account of fetchedAccounts) {
      userSocialMedias.add(account.socialMediaId);
      accountsBySocialMedia.add(account.socialMediaId, {
        ...account,
        valid: false,
      });
    }
` 


deve ser essa:

 `[
      { id: 1, socialMediaId: 'DISCORD_EXAMPLE_ID', valid: false },
      { id: 14, socialMediaId: 'TWITTER_EXAMPLE_ID', valid: false },
    ];` , portanto o trecho abaixo e' resposavel por verificar se o estado `current.accounts.data `  esta com o formato o formato ao lado -> ` `[
      { id: 1, socialMediaId: 'DISCORD_EXAMPLE_ID', valid: false },
      { id: 14, socialMediaId: 'TWITTER_EXAMPLE_ID', valid: false },
    ];` , `